Index: kernel-source/arch/arm/boot/dts/am572x-bel-mmrfic.dts
===================================================================
--- /dev/null
+++ kernel-source/arch/arm/boot/dts/am572x-bel-mmrfic.dts
@@ -0,0 +1,266 @@
+/*
+ * Copyright (C) 2015 PHYTEC America, LLC. - https://www.phytec.com
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ * DTS for MMRFIC kit KPCM-057-SYS
+ */
+
+/dts-v1/;
+
+#include "am572x-phycore-som.dtsi"              /* Enables all SOM features */
+#include "am572x-pcm-948.dtsi"                  /* Carrier Board */
+
+#include "am5728-pcm-057-41300111i.dtsi"        /* SOM variant */
+
+/ {
+        model = "MMRFIC phyCORE-AM572x Carrier Board";
+        compatible = "phytec,am572x-bel-mmrfic", "phytec,am572x-pcm-948",
+                     "phytec,am572x-phycore-som", "ti,am5728", "ti,dra742",
+                     "ti,dra74", "ti,dra7";
+
+
+	rdk_audio: sound {
+                compatible = "simple-audio-card";
+                status = "okay";
+                simple-audio-card,name = "phyCORE-AM57xx-RDK";
+                simple-audio-card,widgets =
+                        "Line", "Line Out",
+                        "Line", "Line In",
+                        "Microphone", "Mic Jack",
+                        "Headphone", "HP Jack";
+                simple-audio-card,routing =
+                        "Line Out",     "LLOUT",
+                        "Line Out",     "RLOUT",
+
+                        "LINE1R",       "Line In",
+                        "LINE1R",       "Line In",
+
+                        "MIC3L",        "Mic Jack",
+                        "MIC3R",        "Mic Jack",
+                        "Mic Jack",     "Mic Bias",
+
+                        "HP Jack",      "HPLOUT",
+                        "HP Jack",      "HPROUT";
+                simple-audio-card,format = "dsp_b";
+                simple-audio-card,bitclock-master = <&sound0_master>;
+                simple-audio-card,frame-master = <&sound0_master>;
+                simple-audio-card,bitclock-inversion;
+
+                simple-audio-card,cpu {
+                        sound-dai = <&mcasp1>;
+                        system-clock-frequency = <12000000>;
+                };
+                sound0_master: simple-audio-card,codec {
+                        sound-dai = <&tlv320aic3x>;
+                        clocks = <&clkout2_clk>;
+                };
+
+	};
+};
+
+
+&dra7_pmx_core {
+        i2c2_pins_default: i2c2_pins_default {
+                pinctrl-single,pins = <
+                DRA7XX_CORE_IOPAD(0x3808, PIN_INPUT_PULLUP | MUX_MODE0)  /* (F17) i2c2_scl */
+                DRA7XX_CORE_IOPAD(0x380c, PIN_INPUT_PULLUP | MUX_MODE0)  /* (C25) i2c2_sda */
+        >;
+        };
+
+        i2c2_pins_sleep: i2c2_pins_sleep {
+                pinctrl-single,pins = <
+                DRA7XX_CORE_IOPAD(0x3808, PIN_INPUT | MUX_MODE15)
+                DRA7XX_CORE_IOPAD(0x380c, PIN_INPUT | MUX_MODE15)
+        >;
+        };
+
+        i2c5_pins_default: i2c5_pins_default {
+                pinctrl-single,pins = <
+                DRA7XX_CORE_IOPAD(0x36B8, PIN_INPUT_PULLUP | MUX_MODE10)  /* (F12) mcasp1_axr1.i2c5_scl */
+                DRA7XX_CORE_IOPAD(0x36B4, PIN_INPUT_PULLUP | MUX_MODE10)  /* (G12) mcasp1_axr0.i2c5_sda */
+        >;
+        };
+
+        i2c5_pins_sleep: i2c5_pins_sleep {
+                pinctrl-single,pins = <
+                DRA7XX_CORE_IOPAD(0x36B8, PIN_INPUT | MUX_MODE15)
+                DRA7XX_CORE_IOPAD(0x34B4, PIN_INPUT | MUX_MODE15)
+        >;
+        };
+        uart1_pins_default:uart1_pins_default {
+                pinctrl-single,pins = <
+                DRA7XX_CORE_IOPAD(0x3734, PIN_OUTPUT | MUX_MODE3)   /* (C18) mcasp4_aclkx.uart8_rxd */
+                DRA7XX_CORE_IOPAD(0x3738, PIN_OUTPUT | MUX_MODE3)   /* (A21) mcasp4_fsx.uart8_txd */
+                DRA7XX_CORE_IOPAD(0x373c, PIN_OUTPUT | MUX_MODE3)   /* (G16) mcasp4_axr0.uart8_ctsn */
+                DRA7XX_CORE_IOPAD(0x3740, PIN_OUTPUT | MUX_MODE3)   /* (D17) mcasp4_axr1.uart8_rtsn */
+        >;
+        };
+
+        uart1_pins_sleep:uart1_pins_sleep {
+                pinctrl-single,pins = <
+                DRA7XX_CORE_IOPAD(0x3734, PIN_OUTPUT | MUX_MODE15)   
+                DRA7XX_CORE_IOPAD(0x3738, PIN_OUTPUT | MUX_MODE15)   
+                DRA7XX_CORE_IOPAD(0x373c, PIN_OUTPUT | MUX_MODE15)  
+                DRA7XX_CORE_IOPAD(0x3740, PIN_OUTPUT | MUX_MODE15) 
+        >;
+        };
+
+        uart2_pins_default:uart2_pins_default {
+                pinctrl-single,pins = <
+                //DRA7XX_CORE_IOPAD(0x3784, PIN_INPUT_SLEW | MUX_MODE2 )   /* (AC7) mmc3_dat0.uart5_rxd */
+                //DRA7XX_CORE_IOPAD(0x3788, PIN_INPUT_SLEW | MUX_MODE2 )   /* (AC6) mmc3_dat1.uart5_txd */
+                DRA7XX_CORE_IOPAD(0x3790, PIN_INPUT_SLEW | MUX_MODE2 )   /* (AC3) mmc3_dat3.uart5_rts */
+                DRA7XX_CORE_IOPAD(0x378c, PIN_INPUT_SLEW | MUX_MODE2 )   /* (AA3) mmc3_dat2_uart5_cts */
+
+        >;
+        };
+
+        uart2_pins_sleep:uart2_pins_sleep {
+                pinctrl-single,pins = <
+                DRA7XX_CORE_IOPAD(0x3784, PIN_INPUT | MUX_MODE15 )   
+                DRA7XX_CORE_IOPAD(0x3788, PIN_INPUT | MUX_MODE15 )   
+                DRA7XX_CORE_IOPAD(0x3790, PIN_INPUT | MUX_MODE15 )   
+                DRA7XX_CORE_IOPAD(0x378c, PIN_INPUT | MUX_MODE15 )   
+
+        >;
+        };
+
+        uart3_pins_default:uart3_pins_default {
+                pinctrl-single,pins = <
+                DRA7XX_CORE_IOPAD(0x37b8, PIN_INPUT_PULLDOWN | MUX_MODE1)      /* (B21) spi1_cs2.uart4_rxd */
+                DRA7XX_CORE_IOPAD(0x37bc, PIN_INPUT_SLEW | MUX_MODE1)           /* (B20) spi1_cs3.uart4_txd */
+        >;
+        };
+
+        uart4_pins_default:uart4_pins_default {
+                pinctrl-single,pins = <
+                DRA7XX_CORE_IOPAD(0x34e4, PIN_INPUT_PULLDOWN | MUX_MODE5)      /* (AD9) vin1a_de0.uart7_rxd */
+                DRA7XX_CORE_IOPAD(0x34e8, PIN_INPUT_SLEW | MUX_MODE5)           /* (AF9) vin1a_de0.uart7_txd */
+        >;
+        };
+
+        uart5_pin:uart {
+                pinctrl-single,pins = <
+                DRA7XX_CORE_IOPAD(0x3648, PIN_INPUT_SLEW | MUX_MODE0)   /* (V2) uart3_rxd */
+                DRA7XX_CORE_IOPAD(0x364c, PIN_INPUT_SLEW | MUX_MODE0)   /* (AA4) uart3_txd */
+                /* DRA7XX_CORE_IOPAD(0x37cc, PIN_INPUT_SLEW | MUX_MODE1)*/    /* (B24) spi2_cs0.uart3_rts */
+                /* DRA7XX_CORE_IOPAD(0x37c8, PIN_INPUT_SLEW | MUX_MODE1) */   /* (AA4) spi2_d0.uart3_cts */
+
+        >;
+        };
+
+        uart6_pins_default:uart6_pins_default {
+                pinctrl-single,pins = <
+                DRA7XX_CORE_IOPAD(0x3570, PIN_INPUT_SLEW | MUX_MODE8)   /*  vin2a_d2.uart10_rxd */
+                DRA7XX_CORE_IOPAD(0x3574, PIN_INPUT_SLEW | MUX_MODE8)   /*    vin2a_d3.uart10_txd */
+        >;
+        };
+
+        uart7_pins_default:uart7_pins_default {
+                pinctrl-single,pins = <
+                DRA7XX_CORE_IOPAD(0x3744, PIN_OUTPUT | MUX_MODE3 )   /* (AA3) mcasp5_aclkx.uart9_rxd */
+                DRA7XX_CORE_IOPAD(0x3748, PIN_OUTPUT | MUX_MODE3 )   /* (AB9) mcasp5_fsx.uart9_txd */
+        >;
+        };
+
+};	
+
+&i2c3 {
+	tlv320aic3x: tlv320aic3x@18 {
+		#sound-dai-cells = <0>;
+                compatible = "ti,tlv320aic3x";
+                reg = <0x18>;
+                pinctrl-names = "default", "sleep";
+                pinctrl-0 = <&clkout2_pins_default>;
+                pinctrl-1 = <&clkout2_pins_sleep>;
+
+                status = "okay";
+                ai3x-micbias-vg = <2>;
+                adc-settle-ms = <40>;
+
+                AVDD-supply = <&vdd_3v3>;
+                IOVDD-supply = <&vdd_3v3>;
+                DRVDD-supply = <&vdd_3v3>;
+                DVDD-supply = <&aic_dvdd>;
+        };
+
+};
+
+&i2c2 {
+        status = "okay";
+        pinctrl-names = "default", "sleep";
+        pinctrl-0 = <&i2c2_pins_default>;
+        pinctrl-1 = <&i2c2_pins_sleep>;
+        clock-frequency = <400000>;
+};
+
+&i2c5 {
+        status = "okay";
+        pinctrl-names = "default", "sleep";
+        pinctrl-0 = <&i2c5_pins_default>;
+        pinctrl-1 = <&i2c5_pins_sleep>;
+        clock-frequency = <400000>;
+};
+
+&i2c4 {
+        tlv320aic3007: tlv320aic3007@18 {
+                status = "disabled";
+        };
+};
+
+&uart1 {
+        status = "okay";
+        pinctrl-0 = <&uart1_pins_default>;
+};
+
+&uart2 {
+        status = "okay";
+        pinctrl-0 = <&uart2_pins_default  &uart5_pins_default>;
+};
+
+&uart3 {
+        status = "okay";
+        pinctrl-0 = <&uart3_pins_default>;
+};
+
+&uart4 {
+        status = "okay";
+        pinctrl-0 = <&uart4_pins_default>;
+};
+
+&uart5 {
+        status = "okay";
+        pinctrl-0 = <&uart5_pins_default>;
+};
+
+&uart6 {
+        status = "okay";
+        pinctrl-0 = <&uart6_pins_default>;
+};
+
+&uart7 {
+        status = "okay";
+        pinctrl-0 = <&uart7_pins_default>;
+};
+
+&hdmi {
+        status = "disabled";
+};
+
+&mmc1 {
+	disable-wp;
+};
+
+&pcie1_rc {
+	status = "disabled";
+};
+
+&mcspi1 {
+	spidev1_0: spidev1@0 {
+		compatible = "rohm,dh2228fv";
+	};
+};
+
